sql/src/main/java/io/crate/operation/collect/collectors/OrderedDocCollector.java

```diff
@@ -120,6 +120,10 @@ public OrderedDocCollector(SearchContext searchContext,
         if (exhausted) {
             return empty;
         }
+        if (batchSize == 0) {
```

Comment:
I think the planner should create a NoopPlan if the limit is 0 and then this could be changed to an assertion in the constructor that the batchSize is never 0